packages/graph/hash_graph/lib/graph/src/shared/subgraph/vertices/mod.rs

```diff
@@ -2,43 +2,70 @@ mod vertex;
 
 use std::collections::{hash_map::Entry, HashMap};
 
-use serde::Serialize;
-use type_system::uri::BaseUri;
-use utoipa::{openapi, ToSchema};
+use serde::{Serialize, Serializer};
 
 pub use self::vertex::*;
 use crate::identifier::{
-    knowledge::{EntityId, EntityVersion},
-    ontology::OntologyTypeVersion,
-    GraphElementEditionId,
+    knowledge::EntityEditionId, ontology::OntologyTypeEditionId, GraphElementEditionId,
 };
 
-#[derive(Default, Debug, Serialize, ToSchema)]
-#[serde(transparent)]
-pub struct OntologyVertices(pub HashMap<BaseUri, HashMap<OntologyTypeVersion, OntologyVertex>>);
+#[derive(Default, Debug)]
+pub struct Vertices {
+    ontology: HashMap<OntologyTypeEditionId, OntologyVertex>,
+    knowledge_graph: HashMap<EntityEditionId, KnowledgeGraphVertex>,
+}
 
-#[derive(Default, Debug, Serialize, ToSchema)]
-#[serde(transparent)]
-pub struct KnowledgeGraphVertices(
-    // TODO: expose it through methods instead of making this field `pub`
-    //   see https://app.asana.com/0/1202805690238892/1203358665695491/f
-    pub HashMap<EntityId, HashMap<EntityVersion, KnowledgeGraphVertex>>,
-);
+impl Serialize for Vertices {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        crate::api::utoipa::subgraph::Vertices {
+            ontology: self
+                .ontology
+                .iter()
+                .fold(HashMap::new(), |mut map, (id, vertex)| {
+                    match map.entry(id.base_id()) {
+                        Entry::Occupied(mut entry) => {
+                            entry.get_mut().insert(id.version(), vertex);
+                        }
+                        Entry::Vacant(entry) => {
+                            let mut map = HashMap::new();
+                            map.insert(id.version(), vertex);
+                            entry.insert(map);
+                        }
+                    }
+                    map
+                }),
+            knowledge_graph: self.knowledge_graph.iter().fold(
+                HashMap::new(),
+                |mut map, (id, vertex)| {
+                    match map.entry(id.base_id()) {
+                        Entry::Occupied(mut entry) => {
+                            entry.get_mut().insert(id.version(), vertex);
+                        }
+                        Entry::Vacant(entry) => {
+                            let mut map = HashMap::new();
+                            map.insert(id.version(), vertex);
+                            entry.insert(map);
+                        }
+                    }
+                    map
+                },
+            ),
+        }
+        .serialize(serializer)
+    }
+}
 
-#[derive(Default, Debug, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct Vertices {
-    #[serde(flatten)]
-    ontology: OntologyVertices,
-    #[serde(flatten)]
-    knowledge_graph: KnowledgeGraphVertices,
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub enum VertexIdentifier {
```

Comment:
Why does this exist

Reply:
I have no idea 🙃 
Fixed in aeecf4a9275ce0d97cdf3f7071711aaa13b4ed4f